src/edu/stanford/nlp/ie/crf/CRFClassifier.java

```diff
@@ -959,42 +917,26 @@ protected void makeAnswerArraysAndTagIndex(Collection<List<IN>> ob) {
    * @param featureFactories The FeatureFactories to use to extract features
    * @return The constructed CRFDatum
    */
-  public CRFDatum<List<String>, CRFLabel> makeDatum(List<IN> info, int loc,
+  public CRFDatum<Collection<String>, CRFLabel> makeDatum(List<IN> info, int loc,
                                                     List<FeatureFactory<IN>> featureFactories) {
     // pad.set(CoreAnnotations.AnswerAnnotation.class, flags.backgroundSymbol); // cdm: isn't this unnecessary, as this is how it's initialized in AbstractSequenceClassifier.reinit?
     PaddedList<IN> pInfo = new PaddedList<>(info, pad);
 
-    ArrayList<List<String>> features = new ArrayList<>();
-    List<double[]> featureVals = new ArrayList<>();
-
-    // for (int i = 0; i < windowSize; i++) {
-    // List featuresC = new ArrayList();
-    // for (int j = 0; j < FeatureFactory.win[i].length; j++) {
-    // featuresC.addAll(featureFactory.features(info, loc,
-    // FeatureFactory.win[i][j]));
-    // }
-    // features.add(featuresC);
-    // }
+    ArrayList<Collection<String>> features = new ArrayList<>(windowSize);
+    List<double[]> featureVals = flags.useEmbedding ? new ArrayList<>(1) : null;
 
-    // todo [cdm Aug 2012]: Since getCliques returns all cliques within its bounds, can't the for loop here be eliminated? But my first attempt to removed failed to produce identical results....
```

Comment:
Any reason for dropping this comment?  The loop is still here, just pushed into eachClique

Reply:
I think the comment refers to the outer loop, over `i`.
And even when pushing this loop into getCliques, it would still be a for loop there, wouldn't it?
At some point you'll want all features at all positions. So I concluded it cannot be removed.